android/rustls-platform-verifier/src/main/java/org/rustls/platformverifier/CertificateVerifier.kt

```diff
@@ -253,6 +253,30 @@ internal object CertificateVerifier {
         val validChain = try {
             trustManager.checkServerTrusted(certificateChain.toTypedArray(), authMethod, serverName)
         } catch (e: CertificateException) {
+            // In test configurations we may see `checkServerTrusted` fail once vendored test
+            // certificates pass their expiry date. We try to avoid that by using a fixed
+            // verification time when calling `endEntity.checkValidity` above, however we can't
+            // fix the time for the `checkServerTrusted` call.
+            //
+            // To make diagnosing CI test failures easier we try to find the root cause of
+            // checkServerTrusted failing, returning a different `StatusCode` as appropriate.
+            if (BuildConfig.TEST) {
+                var rootCause: Throwable? = e
+                while (rootCause?.cause != null && rootCause.cause != rootCause) {
+                    rootCause = rootCause.cause
+                }
```

Comment:
I cribbed this from [this baeldung post](https://www.baeldung.com/java-exception-root-cause), and then tried my best as a non-Kotlin programmer to make it Kotlin-ish :laughing:

Reply:
I believe this can be slightly cleaned up to make it more clear `rootCause` will never actually be `null`:
```diff
diff --git a/android/rustls-platform-verifier/src/main/java/org/rustls/platformverifier/CertificateVerifier.kt b/android/rustls-platform-verifier/src/main/java/org/rustls/platformverifier/CertificateVerifier.kt
index 79421ed..0290956 100644
--- a/android/rustls-platform-verifier/src/main/java/org/rustls/platformverifier/CertificateVerifier.kt
+++ b/android/rustls-platform-verifier/src/main/java/org/rustls/platformverifier/CertificateVerifier.kt
@@ -261,9 +261,11 @@ internal object CertificateVerifier {
             // To make diagnosing CI test failures easier we try to find the root cause of
             // checkServerTrusted failing, returning a different `StatusCode` as appropriate.
             if (BuildConfig.TEST) {
-                var rootCause: Throwable? = e
-                while (rootCause?.cause != null && rootCause.cause != rootCause) {
-                    rootCause = rootCause.cause
+                var rootCause: Throwable = e
+                while (rootCause.cause != rootCause) {
+                    rootCause.cause?.let {
+                        rootCause = it
+                    }
                 }
                 return when (rootCause) {
                     is CertificateExpiredException, is CertificateNotYetValidException -> VerificationResult(
```

Reply:
Good call, fixed. Thanks!

Reply:
I had to revert this change because it causes an infinite loop that binds up the test suite :skull: 

CI has been running the Android test suite with this commit applied for 1h+